src/assertions/overlap.js

```diff
@@ -0,0 +1,30 @@
+import Rect from '../utils/rect';
+
+const linesOverlap = (line1, line2) => {
+    const line2isBefore = line2.end < line1.start;
+    const line2isAfter = line2.start > line1.end;
+    return !(line2isBefore || line2isAfter);
+};
```

Comment:
I'm not sure if this is expected behaviour or a _bug_ but perhaps this should be **>=** and **<=** as is if you have 2 elements naturally side-by-side or above-below with no spacing they flag up as overlapping even though I'm pretty sure they are not?

Reproduce:

```html
<div class="container">
  <div class="block"></div>
  <div class="block"></div>
</div>
```

```css
.container {
  font-size: 0;
}

.block {
  width: 100px;
  height: 100px;
  background: orange;
  display: inline-block;
}

.block + .block {
  background: red;
}
```

Reply:
Good catch and need to add a test case for this. I don't actually know what is correct here as from a purely numerical standpoint, they're overlapping by one pixel as the left one's right == right one's left. So in that sense the linesOverlap function is working as intended. But from a practical standpoint, it wouldn't make a sense to say they overlap, so let's got with that?

Btw, do you want review changes as a separate commit? 
<img width="505" alt="Näyttökuva 2020-6-19 kello 12 47 52" src="https://user-images.githubusercontent.com/526499/85120075-63726280-b22b-11ea-8b91-2f17fbecd79c.png">

Reply:
> So in that sense the linesOverlap function is working as intended. But from a practical standpoint, it wouldn't make a sense to say they overlap, **so let's got with that?**

I agree... I think we should go with this and it can be changed in future should any issues arise from this! I'm happy to review this in a separate pull request!

Great work!

Reply:
Here's the fix https://github.com/msmps/cypress-layout-inspector/pull/10

Reply:
added more tests too for various layout methods (there's one that tests your example above)